packages/react-import-remote/src/ImportRemote.tsx

```diff
@@ -21,91 +14,48 @@ export interface Props<Imported = any> {
   defer?: DeferTiming;
 }
 
-interface State {
-  loaded: boolean;
-  loading: boolean;
-}
-
-export default class ImportRemote extends React.PureComponent<Props, State> {
-  state: State = {loaded: false, loading: false};
-  private idleCallbackHandle?: RequestIdleCallbackHandle;
-
-  componentWillUnmount() {
-    if (this.idleCallbackHandle != null && 'cancelIdleCallback' in window) {
-      (window as WindowWithRequestIdleCallback).cancelIdleCallback(
-        this.idleCallbackHandle,
-      );
-    }
-  }
-
-  async componentDidMount() {
-    const {defer = DeferTiming.Mount} = this.props;
-
-    if (defer === DeferTiming.Idle) {
-      if ('requestIdleCallback' in window) {
-        this.idleCallbackHandle = (window as WindowWithRequestIdleCallback).requestIdleCallback(
-          this.loadRemote,
-        );
-      } else {
-        this.loadRemote();
+export default function ImportRemote({
+  source,
+  nonce,
+  preconnect,
+  onError,
+  getImport,
+  onImported,
+  defer,
+}: Props) {
+  const {result, intersectionRef} = useImportRemote(source, {
+    defer,
+    nonce,
+    getImport,
+  });
+
+  const intersectionObserver =
+    defer === DeferTiming.InViewport && intersectionRef ? (
+      <div ref={intersectionRef} />
+    ) : null;
+
+  React.useEffect(
+    () => {
+      switch (result.status) {
```

Comment:
Hmm I feels weird to me that result is a nested object. I would have expected something more like:

```ts
{status: Status.Complete, imported: T, intersectionRef: Ref} | {status: Status.Error, error: Error, intersectionRef: Ref}
```

Reply:
The way I have it groups things that change together.